### ISSUE TITLE
Internal Server Error when click on JoinSign

### ISSUE THREAD

**Issue by JustJamezX** [thebigsmileXD/Spleef#7]
**CONSOLE MESSAGE:**

```
[03:52:21] [Server thread/CRITICAL]: TypeError: "Argument 1 passed to pocketmine\network\mcpe\NetworkBinaryStream::putString() must be of the type string, int given, called in phar:///home/container/PocketMine-MP.phar/src/pocketmine/network/mcpe/protocol/AddActorPacket.php on line 211" (EXCEPTION) in "src/pocketmine/network/mcpe/NetworkBinaryStream" at line 62
[03:52:21] [Server thread/DEBUG]: #0 src/pocketmine/network/mcpe/protocol/AddActorPacket(211): pocketmine\network\mcpe\NetworkBinaryStream->putString(integer 37)
[03:52:21] [Server thread/DEBUG]: #1 src/pocketmine/network/mcpe/protocol/DataPacket(127): pocketmine\network\mcpe\protocol\AddActorPacket->encodePayload()
[03:52:21] [Server thread/DEBUG]: #2 src/pocketmine/network/mcpe/RakLibInterface(244): pocketmine\network\mcpe\protocol\DataPacket->encode()
[03:52:21] [Server thread/DEBUG]: #3 src/pocketmine/Player(3283): pocketmine\network\mcpe\RakLibInterface->putPacket(object pocketmine\Player, object pocketmine\network\mcpe\protocol\AddActorPacket, boolean , boolean )
[03:52:21] [Server thread/DEBUG]: #4 src/pocketmine/Player(3300): pocketmine\Player->sendDataPacket(object pocketmine\network\mcpe\protocol\AddActorPacket, boolean , boolean )
[03:52:21] [Server thread/DEBUG]: #5 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/libs/xenialdan/gameapi/libs/xenialdan/apibossbar/DiverseBossBar(187): pocketmine\Player->dataPacket(object pocketmine\network\mcpe\protocol\AddActorPacket)
[03:52:21] [Server thread/DEBUG]: #6 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/libs/xenialdan/gameapi/libs/xenialdan/apibossbar/BossBar(92): xenialdan\Spleef\libs\xenialdan\gameapi\libs\xenialdan\apibossbar\DiverseBossBar->sendSpawnPacket(array[1])
[03:52:21] [Server thread/DEBUG]: #7 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/libs/xenialdan/gameapi/libs/xenialdan/apibossbar/DiverseBossBar(46): xenialdan\Spleef\libs\xenialdan\gameapi\libs\xenialdan\apibossbar\BossBar->addPlayer(object pocketmine\Player)
[03:52:21] [Server thread/DEBUG]: #8 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/libs/xenialdan/gameapi/Arena(221): xenialdan\Spleef\libs\xenialdan\gameapi\libs\xenialdan\apibossbar\DiverseBossBar->addPlayer(object pocketmine\Player)
[03:52:21] [Server thread/DEBUG]: #9 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/JoinGameListener(40): xenialdan\Spleef\libs\xenialdan\gameapi\Arena->joinTeam(object pocketmine\Player)
[03:52:21] [Server thread/DEBUG]: #10 plugins/Spleef_dev-2.phar/src/xenialdan/Spleef/JoinGameListener(26): xenialdan\Spleef\JoinGameListener->onClickSign(object pocketmine\event\player\PlayerInteractEvent, array[4])
[03:52:21] [Server thread/DEBUG]: #11 src/pocketmine/plugin/MethodEventExecutor(42): xenialdan\Spleef\JoinGameListener->onInteract(object pocketmine\event\player\PlayerInteractEvent)
[03:52:21] [Server thread/DEBUG]: #12 src/pocketmine/plugin/RegisteredListener(80): pocketmine\plugin\MethodEventExecutor->execute(object xenialdan\Spleef\JoinGameListener, object pocketmine\event\player\PlayerInteractEvent)
[03:52:21] [Server thread/DEBUG]: #13 src/pocketmine/event/Event(88): pocketmine\plugin\RegisteredListener->callEvent(object pocketmine\event\player\PlayerInteractEvent)
[03:52:21] [Server thread/DEBUG]: #14 src/pocketmine/level/Level(1870): pocketmine\event\Event->call()
[03:52:21] [Server thread/DEBUG]: #15 src/pocketmine/Player(2491): pocketmine\level\Level->useItemOn(object pocketmine\math\Vector3, object pocketmine\item\ItemBlock, integer 5, object pocketmine\math\Vector3, object pocketmine\Player, boolean 1)
[03:52:21] [Server thread/DEBUG]: #16 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(150): pocketmine\Player->handleInventoryTransaction(object pocketmine\network\mcpe\protocol\InventoryTransactionPacket)
[03:52:21] [Server thread/DEBUG]: #17 src/pocketmine/network/mcpe/protocol/InventoryTransactionPacket(171): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleInventoryTransaction(object pocketmine\network\mcpe\protocol\InventoryTransactionPacket)
[03:52:21] [Server thread/DEBUG]: #18 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(110): pocketmine\network\mcpe\protocol\InventoryTransactionPacket->handle(object pocketmine\network\mcpe\PlayerNetworkSessionAdapter)
[03:52:21] [Server thread/DEBUG]: #19 src/pocketmine/network/mcpe/protocol/BatchPacket(127): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object pocketmine\network\mcpe\protocol\InventoryTransactionPacket)
[03:52:21] [Server thread/DEBUG]: #20 src/pocketmine/network/mcpe/PlayerNetworkSessionAdapter(110): pocketmine\network\mcpe\protocol\BatchPacket->handle(object pocketmine\network\mcpe\PlayerNetworkSessionAdapter)
[03:52:21] [Server thread/DEBUG]: #21 src/pocketmine/Player(3235): pocketmine\network\mcpe\PlayerNetworkSessionAdapter->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[03:52:21] [Server thread/DEBUG]: #22 src/pocketmine/network/mcpe/RakLibInterface(169): pocketmine\Player->handleDataPacket(object pocketmine\network\mcpe\protocol\BatchPacket)
[03:52:21] [Server thread/DEBUG]: #23 vendor/pocketmine/raklib/src/server/ServerHandler(99): pocketmine\network\mcpe\RakLibInterface->handleEncapsulated(string[21] 112.198.233.110 44628, object raklib\protocol\EncapsulatedPacket, integer 0)
[03:52:21] [Server thread/DEBUG]: #24 src/pocketmine/network/mcpe/RakLibInterface(109): raklib\server\ServerHandler->handlePacket()
[03:52:21] [Server thread/DEBUG]: #25 src/pocketmine/network/mcpe/RakLibInterface(99): pocketmine\network\mcpe\RakLibInterface->process()
[03:52:21] [Server thread/DEBUG]: #26 vendor/pocketmine/snooze/src/SleeperHandler(123): pocketmine\network\mcpe\RakLibInterface->pocketmine\network\mcpe\{closure}()
[03:52:21] [Server thread/DEBUG]: #27 vendor/pocketmine/snooze/src/SleeperHandler(85): pocketmine\snooze\SleeperHandler->processNotifications()
[03:52:21] [Server thread/DEBUG]: #28 src/pocketmine/Server(2157): pocketmine\snooze\SleeperHandler->sleepUntil(double 1593834741.3186)
[03:52:21] [Server thread/DEBUG]: #29 src/pocketmine/Server(1994): pocketmine\Server->tickProcessor()
[03:52:21] [Server thread/DEBUG]: #30 src/pocketmine/Server(1588): pocketmine\Server->start()
[03:52:21] [Server thread/DEBUG]: #31 src/pocketmine/PocketMine(273): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[16] /home/container/, string[24] /home/container/plugins/)
[03:52:21] [Server thread/DEBUG]: #32 src/pocketmine/PocketMine(304): pocketmine\server()
[03:52:21] [Server thread/DEBUG]: #33 (11): require(string[71] phar:///home/container/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
object(xenialdan\Spleef\libs\xenialdan\gameapi\Team)#14102 (7) {
["players":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
array(1) {
["jamescool5757"]=>
object(pocketmine\Player)#13504 (0) {
["initialplayers":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
array(0) {
["max":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
int(16)
["min":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
int(2)
["name":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
string(7) "Players"
["color":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
string(3) "§r"
["spawn":"xenialdan\Spleef\libs\xenialdan\gameapi\Team":private]=>
NULL
```

I just clicked on the joinsign with the format:

```
L1: [Spleef]
L2: Spleef-1
L3: 
L4: 
```
If theres a fix please tell me, TY!

**Comment by inxomnyaa** [comment]
It appears that apibossbar is not up to date in this plugin. I will look into it.

**Comment by inxomnyaa** [comment]
Possibly fixed in 88d74d9
